<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">  
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    
    <title>Comparing Annotations - Bioinformatics course</title>
    

    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/3.003/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="../../../css/base.min.css" rel="stylesheet">
    <link href="../../../css/cinder.min.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.min.css">
    <link href="../../../css/admonition.css" rel="stylesheet">
    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
            <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
        <![endif]-->

    <script src="//ajax.googleapis.com/ajax/libs/webfont/1.6.28/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    

     
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">First week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../blast/blast_online/">Blast Online</a>
</li>

                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../blast/blast_cli/">Command-line Blast</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Second week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/rna/">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/wms/">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 5 - Eukaryotic Annotation <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../schedule/">Schedule</a>
</li>

                        
                            
<li >
    <a href="../practical1/">Practical 1</a>
</li>

                        
                            
<li >
    <a href="../practical2/">Practical 2</a>
</li>

                        
                            
<li >
    <a href="../practical3_manualCuration/">Practical 3</a>
</li>

                        
                            
<li >
    <a href="../practical4_funcAnnotInterp/">Practical 4</a>
</li>

                        
                            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Extras</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../UsingWebApollo.md">WebApollo</a>
</li>

        
            
<li >
    <a href="../practical2_sub_gatherEvidence/">Protein, EST and RNA-seq data</a>
</li>

        
            
<li >
    <a href="../practical2_sub_transcriptome/">Assembling transcripts</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerNoAbinit/">Evidence Based Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerAbinit/">Abinitio Annotation</a>
</li>

        
            
<li class="active">
    <a href="./">Comparing Annotations</a>
</li>

        
            
<li >
    <a href="../practical2_supl_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl3_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl2_maker/">Inspecting Maker output</a>
</li>

        
    </ul>
  </li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fas fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../practical2_sub_makerAbinit/">
                            <i class="fas fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../practical2_supl_maker/">
                            Next <i class="fas fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#comparing-and-evaluating-annotations">Comparing and evaluating annotations</a></li>
            <li class="second-level"><a href="#evaluating-annotation-quality">Evaluating annotation quality</a></li>
                
                <li class="third-level"><a href="#busco">BUSCO</a></li>
            <li class="second-level"><a href="#comparing-annotations">Comparing annotations</a></li>
                
                <li class="third-level"><a href="#preparing-the-input-files">Preparing the input files</a></li>
                <li class="third-level"><a href="#counting-features">Counting features</a></li>
                <li class="third-level"><a href="#pairwise-comparison-of-features">Pairwise comparison of features</a></li>
            <li class="second-level"><a href="#visualising-annotations">Visualising annotations</a></li>
                
                <li class="third-level"><a href="#using-webapollo-to-view-annotations">Using WebApollo to view annotations</a></li>
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="comparing-and-evaluating-annotations">Comparing and evaluating annotations</h1>
<p>In this exercise you will take the three annotations you have created for Drosophila - the pure abinitio one done with augustus, the evidence-based done with MAKER and the abinitio evidence drived one done with MAKER. First, we will count the features annotated in each of them and compare that number against the existing reference annotation. Next, we will perform a proper feature-level comparison to obtain a proper estimate of congruency.</p>
<h2 id="evaluating-annotation-quality">Evaluating annotation quality</h2>
<p>Evaluating an annotation can be done in three ways - running busco but with the proteins obtained from the annotation, in comparison with another annotation or in reference to the evidence alignments. The former isn't so much a quality check as a measure of congruency - i.e. the resulting numbers don't tell you which of the two gene builds is more correct. On the other hand, a comparison with evidence alignments is what Maker uses internally to select gene models. After synthesizing and annotating loci, the resulting model will be ranked against the filtered evidence alignments. The more congruent these two points of information are, the lower the 'annotation edit distance' (AED) will be. The AED score can be used to e.g. check an annotation for problematic models that may then be subjected to manual curation.</p>
<h3 id="busco">BUSCO</h3>
<p>BUSCO is run before annotating to check if the assembly is good and therefore if the annotation will be good. It is also run after the structural annotation to then compare if we indeed find a number of genes corresponding of the first run of busco.</p>
<p>You will need to link the protein file created by maker on the run with the ab-initio</p>
<pre><code>cd ~/annotation_course/practical2
mkdir busco
cd busco

ln -s ../maker/maker_with_abinitio/annotations.proteins.fa   
ln -s ~/annotation_course/practical1/busco/metazoa_odb9

BUSCO.py -i annotations.proteins.fa -o dmel_maker_abinitio -m prot -c 8 -l metazoa_odb9
</code></pre>

<ul>
<li>if you compare with you first busco results what do you see?</li>
</ul>
<h2 id="comparing-annotations">Comparing annotations</h2>
<p>As with many tasks within bioinformatics, it is always a great idea to first look around for existing solutions. In the case of comparing annotations, there are in fact options already out there. One such example is genometools, which we have briefly used before.</p>
<h3 id="preparing-the-input-files">Preparing the input files</h3>
<p>First you have to be situated in a folder containing the two maker annotations (with and without ab initio) and the augustus annotation. </p>
<pre><code>cd ~/annotation_course/practical2
mkdir compare
cd compare
ln -s ../maker/maker_no_abinitio/annotationByType/maker.gff maker_no_abinitio.gff 
ln -s ../maker/maker_with_abinitio/annotationByType/maker.gff maker_abinitio.gff 
</code></pre>

<p>Then, copy or sym-link the EnsEMBL reference annotation.</p>
<pre><code>ln -s ~/annotation_course/data/annotation/ensembl.chr4.gff
</code></pre>

<p>Now we have to sort any GFF3-formatted annotation in a way that genometools accepts.</p>
<pre><code>sed -i '1i##gff-version 3' maker_no_abinitio.gff
sed -i '1i##gff-version 3' maker_abinitio.gff

gt gff3 -sort maker_no_abinitio.gff &gt; maker_no_abinitio.sorted.gff 
gt gff3 -sort maker_abinitio.gff &gt; maker_abinitio.sorted.gff 
gt gff3 -sort ensembl.chr4.gff &gt; ensembl.sorted.gff 
</code></pre>

<h3 id="counting-features">Counting features</h3>
<p>Next, we get the feature counts for the three annotations and the reference from EnsEMBL:</p>
<pre><code>gt stat maker_no_abinitio.gff
</code></pre>

<p>or</p>
<pre><code>gff3_sp_statistics.pl --gff maker_no_abinitio.gff
</code></pre>

<p>(or whatever you decided to name the file(s). The use of the sorted file or the original one changes nothing here)</p>
<p>As you will note, there are some differences - and of course, this is expected, since we used quite different approaches to generate the two gene builds. EnsEMBL on the other hand is originally imported from FlyBase. Obviously, a lot of manual labor and much more data has been put into the FlyBase annotation - and this highlights a common limitation of any computational pipeline. You will simply never reach the same level of quality and detail as seen in a manually curated reference annotation.</p>
<h3 id="pairwise-comparison-of-features">Pairwise comparison of features</h3>
<p>But feature counts alone can't really give you a clear measure of overlap/differences between any two annotations. In order to properly compare them, we can use another function included in genometools.</p>
<p>With the sorted files, we can now perform a comparison:</p>
<pre><code>gt eval ensembl.sorted.gff maker_no_abinitio.sorted.gff  
</code></pre>

<p>This will create a long list of measures for all relevant sequence features with respect to both the 'sensitivity' and 'specificity' - as a measure of how close the annotation comes to a reference. As a reminder, 'specificity' measures the fraction of a reference overlapping a prediction whereas 'sensitivity' measures the fraction of a prediction overlapping a reference.</p>
<p>Note that the measures employed by genometools function in a all-or-nothing fashion. If the overlap is not 100%, it doesn't count (which is why you are unlikely to find gene-level congruencies between your gene builds and the reference annotation).  </p>
<p>From the comparison of your annotations to the Ensembl annotation, which one <strong>seems</strong> to be the most comprehensive to you ?</p>
<h2 id="visualising-annotations">Visualising annotations</h2>
<p><strong>Note:</strong> The following section overlaps with some of the exercises you have done earlier (comparing augustus predictions against the reference annotation).</p>
<p>In the previous tasks, we have looked at the overlap between different gene builds. While this gives us an indication of how similar two annotations are, <strong>it doesn't really allow us to judge the overall quality and similarity of annotations</strong>. Remember, sensitivity and specificity are 'all-or-nothing' - two annotations may be considered very different, but provide similar information, biologically. By that, we mean that two gene models don't need to be 100% identical in their coordinates to tell the scientist that a gene indeed exists in a given location and what it's product looks like.</p>
<p>We therefore need to visually inspect and compare the gene builds. This is a crucial step in any annotation project - gene build pipelines use a set of defined rules, but human pattern recognition is needed to spot potential systematic errors. For example, a pipeline like Maker will simply take all your input and try to synthesize it into an annotation, but it doesn't do too much checks on the data itself. What if you RNA-seq data is messier than you thought? What if your protein data set includes to many 'predicted' proteins that are in clear conflict with the other data?</p>
<p>There exist a number of 'annotation viewers' - IGV, Argo and Apollo, to name a few. A common choice for annotators is the web-based version of Apollo, WebApollo, mostly for its curation capabilities.</p>
<h3 id="using-webapollo-to-view-annotations">Using WebApollo to view annotations</h3>
<p>Transfer your maker annotation files to your computer using the scp command.<br />
Then, jump to <a href="http://annotation-prod.scilifelab.se:8080/NBIS_course/">WebApollo</a> and upload your annotation track into the genome portal called <strong>drosophila_melanogaster_chr4</strong>. <a href="UsingWebapollo">Here find the WebApollo instruction</a><br />
You can now compare your gene builds against this reference. Some questions to ask yourself:</p>
<ul>
<li>Do my gene builds recover all the genes found in the reference?  </li>
<li>What sort of differences are most common?  </li>
</ul></div>
        
        
    </div>

    <footer class="col-md-12 text-center">
        
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
        
        
    </footer>

    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <!-- <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script> -->
    <script src="../../../js/base.js"></script>
    <script src="../../../search/main.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
