<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">   
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    <title></title>

    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/2.018/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/base.css" rel="stylesheet">
    <link href="../../../css/cinder.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">


    <link href="../../../css/admonition.css" rel="stylesheet">


    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 1 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 2 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../software/">Installing Software</a>
</li>

                        
                            
<li >
    <a href="../../../git/">Git</a>
</li>

                        
                            
<li >
    <a href="../../../blast/">Command-line Blast</a>
</li>

                        
                            
<li >
    <a href="../../../project_organisation/">Project Organisation</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li class="active">
    <a href="./">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../wms/">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fa fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../nanopore/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../16S/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#rna-seq">RNA-Seq</a></li>
            <li class="second-level"><a href="#downloading-the-data">Downloading the data</a></li>
                
            <li class="second-level"><a href="#indexing-transcriptome">Indexing transcriptome</a></li>
                
            <li class="second-level"><a href="#quantify-reads-using-salmon">Quantify reads using salmon</a></li>
                
            <li class="second-level"><a href="#import-read-counts-using-tximport">Import read counts using tximport</a></li>
                
            <li class="second-level"><a href="#differential-expression-using-deseq2">Differential expression using DESeq2</a></li>
                
            <li class="second-level"><a href="#kegg-pathway-analysis">KEGG pathway analysis</a></li>
                
                <li class="third-level"><a href="#thanks">Thanks</a></li>
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="rna-seq">RNA-Seq</h1>
<h2 id="downloading-the-data">Downloading the data</h2>
<p>For this tutorial we will use the test data from <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393">this</a> paper:</p>
<blockquote>
<p>Malachi Griffith<em>, Jason R. Walker, Nicholas C. Spies, Benjamin J. Ainscough, Obi L. Griffith</em>. 2015. Informatics for RNA-seq: A web resource for analysis on the cloud. PLoS Comp Biol. 11(8):e1004393.</p>
</blockquote>
<p>The test data consists of two commercially available RNA samples: Universal Human Reference (UHR) and Human Brain Reference (HBR). The UHR is total RNA isolated from a diverse set of 10 cancer cell lines. The HBR is total RNA isolated from the brains of 23 Caucasians, male and female, of varying age but mostly 60-80 years old.</p>
<p>In addition, a spike-in control was used. Specifically we added an aliquot of the ERCC ExFold RNA Spike-In Control Mixes to each sample. The spike-in consists of 92 transcripts that are present in known concentrations across a wide abundance range (from very few copies to many copies). This range allows us to test the degree to which the RNA-seq assay (including all laboratory and analysis steps) accurately reflects the relative abundance of transcript species within a sample. There are two 'mixes' of these transcripts to allow an assessment of differential expression output between samples if you put one mix in each of your two comparisons. In our case, Mix1 was added to the UHR sample, and Mix2 was added to the HBR sample. We also have 3 complete experimental replicates for each sample. This allows us to assess the technical variability of our overall process of producing RNA-seq data in the lab.</p>
<p>For all libraries we prepared low-throughput (Set A) TruSeq Stranded Total RNA Sample Prep Kit libraries with Ribo-Zero Gold to remove both cytoplasmic and mitochondrial rRNA. Triplicate, indexed libraries were made starting with 100ng Agilent/Strategene Universal Human Reference total RNA and 100ng Ambion Human Brain Reference total RNA. The Universal Human Reference replicates received 2 ul of 1:1000 ERCC Mix 1. The Human Brain Reference replicates received 1:1000 ERCC Mix 2. The libraries were quantified with KAPA Library Quantification qPCR and adjusted to the appropriate concentration for sequencing. The triplicate, indexed libraries were then pooled prior to sequencing. Each pool of three replicate libraries were sequenced across 2 lanes of a HiSeq 2000 using paired-end sequence chemistry with 100bp read lengths.</p>
<p>So to summarize we have:</p>
<ul>
<li>UHR + ERCC Spike-In Mix1, Replicate 1</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 2</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 3</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 1</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 2</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 3</li>
</ul>
<p>You can download the data from <a href="http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar">here</a>.</p>
<p>Download and unpack the data</p>
<pre><code class="bash">curl -O -J -L https://osf.io/7zepj/download
tar xzf toy_rna.tar.gz
cd toy_rna
</code></pre>

<h2 id="indexing-transcriptome">Indexing transcriptome</h2>
<pre><code class="bash">salmon index -t chr22_transcripts.fa -i chr22_index
</code></pre>

<h2 id="quantify-reads-using-salmon">Quantify reads using salmon</h2>
<pre><code class="bash">for i in *_R1.fastq.gz
do
   prefix=$(basename $i _R1.fastq.gz)
   salmon quant -i chr22_index --libType A \
          -1 ${prefix}_R1.fastq.gz -2 ${prefix}_R2.fastq.gz -o quant/${prefix};
done
</code></pre>

<p>This loop simply goes through each sample and invokes salmon using fairly basic options:</p>
<ul>
<li>The -i argument tells salmon where to find the index</li>
<li>--libType A tells salmon that it should automatically determine the library type of the sequencing reads (e.g. stranded vs. unstranded etc.)</li>
<li>The -1 and -2 arguments tell salmon where to find the left and right reads for this sample (notice, salmon will accept gzipped FASTQ files directly).</li>
<li>the -o argument specifies the directory where salmon’s quantification results sould be written.</li>
</ul>
<p>Salmon exposes many different options to the user that enable extra features or modify default behavior.
However, the purpose and behavior of all of those options is beyond the scope of this introductory tutorial.
You can read about salmon’s many options in the <a href="http://salmon.readthedocs.io/en/latest/">documentation</a>.</p>
<p>After the salmon commands finish running, you should have a directory named <code>quant</code>, which will have a sub-directory for each sample.
These sub-directories contain the quantification results of salmon, as well as a lot of other information salmon records about the sample and the run.
The main output file (called quant.sf) is rather self-explanatory. For example, take a peek at the quantification file for sample <code>HBR_Rep1</code> in <code>quant/HBR_Rep1/quant.sf</code> and you’ll see a simple TSV format file listing the name (Name) of each transcript, its length (Length), effective length (EffectiveLength) (more details on this in the documentation), and its abundance in terms of Transcripts Per Million (TPM) and estimated number of reads (NumReads) originating from this transcript.</p>
<h2 id="import-read-counts-using-tximport">Import read counts using tximport</h2>
<p>Using the tximport R package, you can import salmon’s transcript-level quantifications and optionally aggregate them to the gene level for gene-level differential expression analysis.</p>
<p>First, go in Rstudio server by typing the address to your server in your browser:</p>
<p><code>http://MY_IP_ADDRESS:8787/</code></p>
<p>where you replace <code>MY_IP_ADDRESS</code> by the IP address of your Virtual Machine.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To access Rstudio server on the virtual machine, you'll need a password
Ask your instructor for the password!</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you wish, you may work on Rstudio on your own laptop if it is powerful enough.
You will need an up-to-date version of R, and can install the necessary packages using <a href="https://osf.io/a7kqz/download">this script</a></p>
<p>You will also need to download the <code>toy_rna</code> directory</p>
</div>
<p>Once in Rstudio, set your working directory</p>
<pre><code class="R">setwd('~/toy_rna')
</code></pre>

<p>Then load the modules:</p>
<pre><code class="R">library(tximport)
library(GenomicFeatures)
library(readr)
</code></pre>

<p>Salmon did the quantifiation of the transcript level.
We want to see which genes are differentially expressed, so we need to link the transcript names to the gene names.
We can use our .gtf annotation for that, and the GenomicFeatures package:</p>
<pre><code class="R">txdb &lt;- makeTxDbFromGFF(&quot;chr22_genes.gtf&quot;)
k &lt;- keys(txdb, keytype = &quot;GENEID&quot;)
tx2gene &lt;- select(txdb, keys = k, keytype = &quot;GENEID&quot;, columns = &quot;TXNAME&quot;)
head(tx2gene)
</code></pre>

<p>Now we can import the salmon quantification.</p>
<pre><code class="R">samples &lt;- read.table(&quot;samples.txt&quot;, header = TRUE)
files &lt;- file.path(&quot;quant&quot;, samples$sample, &quot;quant.sf&quot;)
names(files) &lt;- paste0(samples$sample)
txi.salmon &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene)
</code></pre>

<p>Take a look at the data:</p>
<pre><code class="R">head(txi.salmon$counts)
</code></pre>

<h2 id="differential-expression-using-deseq2">Differential expression using DESeq2</h2>
<p>load DESeq2:</p>
<pre><code class="R">library(DESeq2)
</code></pre>

<p>Instantiate the DESeqDataSet and generate result table. See <code>?DESeqDataSetFromTximport</code> and <code>?DESeq</code> for more information about the steps performed by the program.</p>
<pre><code class="R">dds &lt;- DESeqDataSetFromTximport(txi.salmon, samples, ~condition)
dds &lt;- DESeq(dds)
res &lt;- results(dds)
</code></pre>

<p>Run the <code>summary</code> command to get an idea of how many genes are up- and downregulated between the two conditions:</p>
<pre><code class="R">summary(res)
</code></pre>

<p>DESeq uses a negative binomial distribution. Such distributions have two parameters: mean and dispersion. The dispersion is a parameter describing how much the variance deviates from the mean.</p>
<p>You can read more about the methods used by DESeq2 in the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">paper</a> or the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq/inst/doc/DESeq.pdf">vignette</a></p>
<p>Plot dispersions:</p>
<pre><code class="R">plotDispEsts(dds, main=&quot;Dispersion plot&quot;)
</code></pre>

<p>For clustering and heatmaps, we need to log transform our data:</p>
<pre><code class="R">rld &lt;- rlogTransformation(dds)
head(assay(rld))
</code></pre>

<p>Then, we create a sample distance heatmap:</p>
<pre><code class="R">library(RColorBrewer)
library(gplots)

(mycols &lt;- brewer.pal(8, &quot;Dark2&quot;)[1:length(unique(samples$condition))])
sampleDists &lt;- as.matrix(dist(t(assay(rld))))
heatmap.2(as.matrix(sampleDists), key=F, trace=&quot;none&quot;,
          col=colorpanel(100, &quot;black&quot;, &quot;white&quot;),
          ColSideColors=mycols[samples$condition],
          RowSideColors=mycols[samples$condition],
          margin=c(10, 10), main=&quot;Sample Distance Matrix&quot;)
</code></pre>

<p>We can also plot a PCA:</p>
<pre><code class="R">DESeq2::plotPCA(rld, intgroup=&quot;condition&quot;)
</code></pre>

<p>It is time to look at some p-values:</p>
<pre><code class="R">table(res$padj&lt;0.05)
res &lt;- res[order(res$padj), ]
resdata &lt;- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=&quot;row.names&quot;, sort=FALSE)
names(resdata)[1] &lt;- &quot;Gene&quot;
head(resdata)
</code></pre>

<p>Examine plot of p-values, the MA plot and the Volcano Plot:</p>
<pre><code class="R">hist(res$pvalue, breaks=50, col=&quot;grey&quot;)
DESeq2::plotMA(dds, ylim=c(-1,1), cex=1)

# Volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main=&quot;Volcano plot&quot;, xlim=c(-2.5,2)))
with(subset(res, padj&lt;.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;red&quot;))
</code></pre>

<h2 id="kegg-pathway-analysis">KEGG pathway analysis</h2>
<p>As always, load the necessary packages:</p>
<pre><code class="R">library(AnnotationDbi)
library(org.Hs.eg.db)
library(pathview)
library(gage)
library(gageData)
</code></pre>

<p>Let’s use the <code>mapIds</code> function to add more columns to the results. The row.names of our results table has the Ensembl gene ID (our key), so we need to specify  <code>keytype=ENSEMBL</code>. The column argument tells the <code>mapIds</code> function which information we want, and the <code>multiVals</code> argument tells the function what to do if there are multiple possible values for a single input value. Here we ask to just give us back the first one that occurs in the database. Let’s get the Entrez IDs, gene symbols, and full gene names.</p>
<pre><code class="R">res$symbol &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)
res$entrez &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;ENTREZID&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)
res$name &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;GENENAME&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)

head(res)
</code></pre>

<p>We’re going to use the <a href="https://bioconductor.org/packages/release/bioc/html/gage.html">gage</a> package for pathway analysis, and the <a href="https://bioconductor.org/packages/release/bioc/html/pathview.html">pathview</a> package to draw a pathway diagram.</p>
<p>The gageData package has pre-compiled databases mapping genes to KEGG pathways and GO terms for common organisms:</p>
<pre><code class="R">data(kegg.sets.hs)
data(sigmet.idx.hs)
kegg.sets.hs &lt;- kegg.sets.hs[sigmet.idx.hs]
head(kegg.sets.hs, 3)
</code></pre>

<p>Run the pathway analysis. See help on the gage function with <code>?gage</code>. Specifically, you might want to try changing the value of same.dir.</p>
<pre><code class="R">foldchanges &lt;- res$log2FoldChange
names(foldchanges) &lt;- res$entrez
keggres &lt;- gage(foldchanges, gsets=kegg.sets.hs, same.dir=TRUE)
lapply(keggres, head)
</code></pre>

<p>Pull out the top 5 upregulated pathways, then further process that just to get the IDs. We’ll use these KEGG pathway IDs downstream for plotting. The <code>dplyr</code> package is required to use the pipe (<code>%&gt;%</code>) construct.</p>
<pre><code class="R">library(dplyr)

# Get the pathways
keggrespathways &lt;- data.frame(id=rownames(keggres$greater), keggres$greater) %&gt;%
  tbl_df() %&gt;%
  filter(row_number()&lt;=5) %&gt;%
  .$id %&gt;%
  as.character()
keggrespathways

# Get the IDs.
keggresids &lt;- substr(keggrespathways, start=1, stop=8)
keggresids
</code></pre>

<p>Finally, the <code>pathview()</code> function in the pathview package makes the plots. Let’s write a function so we can loop through and draw plots for the top 5 pathways we created above.</p>
<pre><code class="R"># Define plotting function for applying later
plot_pathway &lt;- function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species=&quot;hsa&quot;, new.signature=FALSE)

# Unload dplyr since it conflicts with the next line
detach(&quot;package:dplyr&quot;, unload=T)

# plot multiple pathways (plots saved to disk and returns a throwaway list object)
tmp &lt;- sapply(keggresids, function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species=&quot;hsa&quot;))
</code></pre>

<h4 id="thanks">Thanks</h4>
<p>This material was inspired by Stephen Turner's blog post:</p>
<blockquote>
<p>Tutorial: RNA-seq differential expression &amp; pathway analysis with Sailfish, DESeq2, GAGE, and Pathview: http://www.gettinggeneticsdone.com/2015/12/tutorial-rna-seq-differential.html</p>
</blockquote></div>
        
    </div>

    <footer class="col-md-12 text-center">
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
    </footer>

    <script src="../../../js/jquery-1.10.2.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
    <script src="../../../js/base.js"></script>
    <script src="../../../search/require.js"></script>
    <script src="../../../search/search.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
