<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">   
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    <title></title>

    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/2.018/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/base.css" rel="stylesheet">
    <link href="../../../css/cinder.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">


    <link href="../../../css/admonition.css" rel="stylesheet">


    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 1 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 2 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../software/">Installing Software</a>
</li>

                        
                            
<li >
    <a href="../../../git/">Git</a>
</li>

                        
                            
<li >
    <a href="../../../blast/">Command-line Blast</a>
</li>

                        
                            
<li >
    <a href="../../../project_organisation/">Project Organisation</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../rna/">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li class="active">
    <a href="./">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 5 - Eukaryotic Annotation <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../nbis_annotation/schedule/">Schedule</a>
</li>

                        
                            
<li >
    <a href="../../../nbis_annotation/practical_session/practical1/">Practical 1</a>
</li>

                        
                            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2/">Practical 2</a>
</li>

                        
                            
<li >
    <a href="../../../nbis_annotation/practical_session/practical3_manualCuration/">Practical 3</a>
</li>

                        
                            
<li >
    <a href="../../../nbis_annotation/practical_session/practical4_funcAnnotInterp/">Practical 4</a>
</li>

                        
                            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Extras</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../nbis_annotation/practical_session/UsingWebApollo/">WebApollo</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_sub_gatherEvidence/">Protein, EST and RNA-seq data</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_sub_transcriptome/">Assembling transcripts</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_sub_makerNoAbinit/">Evidence Based Annotation</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_sub_makerAbinit/">Abinitio Annotation</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_sub_makerCompareAnnot/">Comparing Annotations</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_supl_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_supl3_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../../../nbis_annotation/practical_session/practical2_supl2_maker/">Inspecting Maker output</a>
</li>

        
    </ul>
  </li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fa fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../meta_assembly/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../../../nbis_annotation/schedule/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#whole-metagenome-sequencin">Whole Metagenome Sequencin</a></li>
            <li class="second-level"><a href="#table-of-contents">Table of Contents</a></li>
                
            <li class="second-level"><a href="#introduction">Introduction</a></li>
                
                <li class="third-level"><a href="#microbiome-used">Microbiome used</a></li>
                <li class="third-level"><a href="#whole-metagenome-sequencing">Whole Metagenome Sequencing</a></li>
            <li class="second-level"><a href="#softwares-required-for-this-tutorial">Softwares Required for this Tutorial</a></li>
                
            <li class="second-level"><a href="#prepare-and-organise-your-working-directory">Prepare and organise your working directory</a></li>
                
            <li class="second-level"><a href="#getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality</a></li>
                
            <li class="second-level"><a href="#taxonomic-classification">Taxonomic Classification</a></li>
                
            <li class="second-level"><a href="#visualization-with-pavian">Visualization with Pavian</a></li>
                
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="whole-metagenome-sequencin">Whole Metagenome Sequencin</h1>
<h3 id="table-of-contents">Table of Contents</h3>
<ul>
<li><a href="#introduction">Introduction</a><ul>
<li><a href="#the-pig-microbiome">The Pig Microbiome</a></li>
<li><a href="#whole-metagenome-sequencing">Whole Metagenome Sequencing</a></li>
</ul>
</li>
<li><a href="#softwares-required-for-this-tutorial">Softwares Required for this Tutorial</a></li>
<li><a href="#getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality</a></li>
<li><a href="#taxonomic-classification">Taxonomic Classification</a></li>
<li><a href="#visualization">Visualization</a></li>
</ul>
<h3 id="introduction">Introduction</h3>
<h4 id="microbiome-used">Microbiome used</h4>
<p>In this tutorial we will compare samples from the Pig Gut Microbiome to samples from the Human Gut Microbiome. Below you'll find a brief description of the two projects:</p>
<p>The Pig Microbiome:</p>
<blockquote>
<p>Pig is a main species for livestock and biomedicine. The pig genome sequence was recently reported. To boost research, we established a catalogue of the genes of the gut microbiome based on faecal samples of 287 pigs from France, Denmark and China. More than 7.6 million non-redundant genes representing 719 metagenomic species were identified by deep metagenome sequencing, highlighting more similarities with the human than with the mouse catalogue. The pig and human catalogues share only 12.6 and 9.3 % of their genes, respectively, but 70 and 95% of their functional pathways. The pig gut microbiota is influenced by gender, age and breed. Analysis of the prevalence of antibiotics resistance genes (ARGs) reflected antibiotics supplementation in each farm system, and revealed that non-antibiotics-fed animals still harbour ARGs. The pig catalogue creates a resource for whole metagenomics-based studies, highly valuable for research in biomedicine and for sustainable knowledge-based pig farming</p>
</blockquote>
<p>The Human Microbiome:</p>
<blockquote>
<p>We are facing a global metabolic health crisis provoked by an obesity epidemic. Here we report the human gut microbial composition in a population sample of 123 non-obese and 169 obese Danish individuals. We find two groups of individuals that differ by the number of gut microbial genes and thus gut bacterial richness. They harbour known and previously unknown bacterial species at different proportions; individuals with a low bacterial richness (23% of the population) are characterized by more marked overall adiposity, insulin resistance and dyslipidaemia and a more pronounced inflammatory phenotype when compared with high bacterial richness individuals. The obese individuals among the former also gain more weight over time. Only a few bacterial species are sufficient to distinguish between individuals with high and low bacterial richness, and even between lean and obese. Our classifications based on variation in the gut microbiome identify subsets of individuals in the general white adult population who may be at increased risk of progressing to adiposity-associated co-morbidities</p>
</blockquote>
<h4 id="whole-metagenome-sequencing">Whole Metagenome Sequencing</h4>
<p>Whole Metagenome sequencing (WMS), or shotgun metagenome sequencing, is a relatively new and powerful sequencing approach that provides insight into community biodiversity and function. On the contrary of Metabarcoding, where only a specific region of the bacterial community (the 16s rRNA) is sequenced, WMS aims at sequencing all the genomic material present in the environment.</p>
<p>The choice of shotgun or 16S approaches is usually dictated by the nature of the studies being conducted. For instance, 16S is well suited for analysis of large number of samples, i.e., multiple patients, longitudinal studies, etc. but offers limited taxonomical and functional resolution. WMS is generally more expensive but offers increased resolution, and allows the discovery of viruses as well as other mobile genetic elements.</p>
<h3 id="softwares-required-for-this-tutorial">Softwares Required for this Tutorial</h3>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a></li>
<li><a href="https://www.r-project.org/">R</a></li>
<li><a href="https://github.com/fbreitwieser/pavian">Pavian</a></li>
</ul>
<h3 id="prepare-and-organise-your-working-directory">Prepare and organise your working directory</h3>
<p>You will first login to your virtual machine using the IP provided by the teachers.
All the exercise will be performed on your VM in the cloud.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When you login with the ssh command, please add the option -X at the end of it to be able to use graphical interface</p>
</div>
<pre><code class="bash">mkdir ~/wms
cd ~/wms
mkdir data
mkdir results
mkdir scripts
</code></pre>

<h3 id="getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality</h3>
<p>As the data were very big, we have prepared performed a downsampling on all 6 datasets (3 pigs and 3 humans).
We will first download and unpack the data.</p>
<pre><code class="bash">cd ~/wms/data
curl -O -J -L https://osf.io/h9x6e/download
tar xvf subset_wms.tar.gz
cd sub_100000
</code></pre>

<p>We'll use FastQC to check the quality of our data.
FastQC should be already installed on your VM, so you need to type</p>
<pre><code>fastqc
</code></pre>

<p>When FastQC has started you can select the fastq file you just downloaded with <code>file -&gt; open</code><br />
What do you think about the quality of the reads? Do they need trimming? Are there still adapters
present? Overrepresented sequences?</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>FastQC can be downloaded and run on a Windows or Linux computer without installation.
It is available <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a></p>
</div>
<p>Alternatively, run fastqc on the command-line:</p>
<pre><code class="bash">fastqc *.fastq
</code></pre>

<p>If the quality appears to be good, it's because it was probably the cleaned reads that were deposited into SRA.
We can directly move to the classification step.</p>
<h3 id="taxonomic-classification">Taxonomic Classification</h3>
<p><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a> is a system for assigning taxonomic labels to short DNA sequences (i.e. reads)<br />
Kraken aims to achieve high sensitivity and high speed by utilizing exact alignments of k-mers and a novel classification algorithm (sic).</p>
<p>In short, kraken uses a new approach with exact k-mer matching to assign taxonomy to short reads. It is <em>extremely</em> fast compared to traditional
approaches (i.e. BLAST).</p>
<p>By default, the authors of kraken built their database based on RefSeq Bacteria, Archaea and Viruses. We'll use it for the purpose of this tutorial.
We will download a shrunk database (minikraken) provided by Kraken developers that is only 4GB.</p>
<pre><code class="bash"># First we create a databases directory in our home
cd /mnt
sudo mkdir databases
cd databases
# Then we download the minikraken database
sudo wget https://ccb.jhu.edu/software/kraken/dl/minikraken_20171019_4GB.tgz
sudo tar xzf minikraken_20171019_4GB.tgz
KRAKEN_DB=/mnt/databases/minikraken_20171013_4GB
cd
</code></pre>

<p>Now run kraken on the reads</p>
<pre><code class="bash"># In the data/ directory
cd ~/wms/data/sub_100000
for i in *_1.fastq
do
    prefix=$(basename $i _1.fastq)
    # print which sample is being processed
    echo $prefix
    kraken --db $KRAKEN_DB --threads 2 --fastq-input \
        ${prefix}_1.fastq ${prefix}_2.fastq &gt; /home/student/wms/results/${prefix}.tab
    kraken-report --db $KRAKEN_DB \
        /home/student/wms/results/${prefix}.tab &gt; /home/student/wms/results/${prefix}_tax.txt
done
</code></pre>

<p>which produces a tab-delimited file with an assigned TaxID for each read.</p>
<p>Kraken includes a script called <code>kraken-report</code> to transform this file into a "tree" view with the percentage of reads assigned to each taxa. We've run this script at each step in the loop. Take a look at the <code>_tax.txt</code> files!</p>
<h3 id="visualization-with-pavian">Visualization with Pavian</h3>
<p>Pavian is a web application for exploring metagenomics classification results.</p>
<p>First, go in Rstudio server by typing the address to your server in your browser:</p>
<p><code>http://MY_IP_ADDRESS:8787/</code></p>
<p>where you replace <code>MY_IP_ADDRESS</code> by the IP address of your Virtual Machine.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To access Rstudio server on the virtual machine, you'll need a password.
Ask your instructor for the password!</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you wish, you may work on Rstudio on your own laptop if it is powerful enough.
You will need an up-to-date version of R, and can install the necessary packages using <a href="https://osf.io/a7kqz/download">this script</a></p>
</div>
<p>Install and run Pavian:</p>
<pre><code class="R">options(repos = c(CRAN = &quot;http://cran.rstudio.com&quot;))
if (!require(remotes)) { install.packages(&quot;remotes&quot;) }
remotes::install_github(&quot;fbreitwieser/pavian&quot;)
pavian::runApp(port=5000)
</code></pre>

<p>Then you will explore and compare the results produced by Kraken.</p></div>
        
    </div>

    <footer class="col-md-12 text-center">
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
    </footer>

    <script src="../../../js/jquery-1.10.2.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
    <script src="../../../js/base.js"></script>
    <script src="../../../search/require.js"></script>
    <script src="../../../search/search.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
