<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">  
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    
    <title>Inspecting Maker output - Bioinformatics course</title>
    

    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/3.003/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="../../../css/base.min.css" rel="stylesheet">
    <link href="../../../css/cinder.min.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.min.css">
    <link href="../../../css/admonition.css" rel="stylesheet">
    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
            <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
        <![endif]-->

    <script src="//ajax.googleapis.com/ajax/libs/webfont/1.6.28/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    

     
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">First week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../blast/blast_online/">Blast Online</a>
</li>

                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../blast/blast_cli/">Command-line Blast</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Second week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../tutorials/docs/rna/">RNA Sequencing</a>
</li>

                        
                            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Eukaryotic Annotation</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../schedule/">Intro</a>
</li>

        
            
<li >
    <a href="../practical1/">Practical 1</a>
</li>

        
            
<li >
    <a href="../practical2/">Practical 2</a>
</li>

        
            
<li >
    <a href="../practical3_manualCuration/">Practical 3</a>
</li>

        
            
<li >
    <a href="../practical4_funcAnnotInterp/">Practical 4</a>
</li>

        
            
<li >
    <a href="../UsingWebApollo.md">WebApollo</a>
</li>

        
            
<li >
    <a href="../practical2_sub_gatherEvidence/">Protein, EST and RNA-seq data</a>
</li>

        
            
<li >
    <a href="../practical2_sub_transcriptome/">Assembling transcripts</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerNoAbinit/">Evidence Based Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerAbinit/">Abinitio Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerCompareAnnot/">Comparing Annotations</a>
</li>

        
            
<li >
    <a href="../practical2_supl_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl3_maker/">Configuring Maker (I)</a>
</li>

        
            
<li class="active">
    <a href="./">Inspecting Maker output</a>
</li>

        
    </ul>
  </li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 5 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../tutorials/docs/16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 6 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../tutorials/docs/meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/wms/">Comparative metagenomics</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/metavir/">Viral metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fas fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../practical2_supl3_maker/">
                            <i class="fas fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../../../tutorials/docs/16S/">
                            Next <i class="fas fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#inspect-the-output">Inspect the output</a></li>
            <li class="second-level"><a href="#finding-your-way-around">Finding your way around</a></li>
                
        <li class="first-level "><a href="#understanding-a-maker-annotation">Understanding a Maker annotation</a></li>
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h2 id="inspect-the-output">Inspect the output</h2>
<h3 id="finding-your-way-around">Finding your way around</h3>
<p>By default, Maker will write the output of its different analyses into a folder named:</p>
<p><strong>&lt;name_of_genome_fasta&gt;.maker.output</strong></p>
<p>In our case:</p>
<p><strong>genome.maker.output</strong></p>
<p>Within the main output directory, Maker keeps a copy of the config files, a database (here: <strong>genome.db</strong>), directories for the blast databases created from your evidence data and a file called <strong>genome_master_datastore_index.log</strong>.</p>
<p>Out of these files, only the <strong>genome_master_datastore_index.log</strong> is really interesting to us. It includes a log of all the contigs included in the genome fasta file - together with their processing status (ideally: FINISHED) and the location of the output files. Since Maker can technically run in parallel on a large number of contigs, it creates separate folders for each of these input data. For larger genomes, this can generate a very deep and confusing folder tree. The <strong>genome_master_datastore_index.log</strong> helps you make sense of it:</p>
<pre><code>4       genome_datastore/A8/7F/4/ STARTED  
4       genome_datastore/A8/7F/4/ FINISHED
</code></pre>

<p>This meens the sequence <strong>4</strong> was started - and finished, with all data (annotation, protein predictions etc) written to the subfolder <strong>genome_datastore/A8/7F/4/</strong>.</p>
<p>If you look into that folder, you will find the finished Maker annotation for this contig.</p>
<pre><code>rw-rw-r- 1 student student 472193 Mar 24 10:16 4.gff
*rw-rw-r- 1 student student 3599 Mar 24 10:16 4.maker.augustus_masked.proteins.fasta
*rw-rw-r- 1 student student 10388 Mar 24 10:16 4.maker.augustus_masked.transcripts.fasta
*rw-rw-r- 1 student student 176 Mar 24 10:16 4.maker.non_overlapping_ab_initio.proteins.fasta 
*rw-rw-r- 1 student student 328 Mar 24 10:16 4.maker.non_overlapping_ab_initio.transcripts.fasta
rw-rw-r- 1 student student 3931 Mar 24 10:16 4.maker.proteins.fasta
rw-rw-r- 1 student student 20865 Mar 24 10:16 4.maker.transcripts.fasta
rw-rw-r- 1 student student 4248 Mar 24 10:15 run.log
drwxrwsr-x 3 student student 4096 Mar 24 10:16 theVoid.4
</code></pre>

<p>* only if an abinitio tool has been activated</p>
<p>The main annotation file is '4.gff' - including both the finished gene models and all the raw compute data. The other files include fasta files for the different sequence features that have been annotated - based on ab-initio predictions through augustus as well as on the finished gene models. The folder 'theVoid' include all the raw computations that Maker has performed to synthesize the evidence into gene models.</p>
<h2 id="understanding-a-maker-annotation">Understanding a Maker annotation</h2>
<p>You have two options now for gathering the output in some usable form - copy select files by hand to wherever you want them. Or you can use a script that does the job for you (we have included an example in the script folder).</p>
<p>From the folder you have run Maker, run the script called 'maker_merge_outputs_from_datastore' to create an output file for all annotations and protein files:</p>
<pre><code>maker_merge_outputs_from_datastore.pl 
</code></pre>

<p>This will create a directory called "<strong>maker_output_processed</strong>" containing a long list of files depending on parameters used for running MAKER.  </p>
<pre><code>augustus_masked.gff
blastn.gff
blastx.gff
cdna2genome.gff
est_gff_est2genome.gff
evm.gff
fgenesh_masked.gff
genemark.gff
genome.all.maker.augustus_masked.proteins.fasta
genome.all.maker.augustus_masked.transcripts.fasta
genome.all.maker.evm.proteins.fasta
genome.all.maker.evm.transcripts.fasta
genome.all.maker.fgenesh.proteins.fasta
genome.all.maker.fgenesh.transcripts.fasta
genome.all.maker.genemark.proteins.fasta
genome.all.maker.genemark.transcripts.fasta
genome.all.maker.noncoding.fasta
genome.all.maker.non_overlapping_ab_initio.proteins.fasta
genome.all.maker.non_overlapping_ab_initio.transcripts.fasta
genome.all.maker.proteins.fasta
genome.all.maker.snap_masked.proteins.fasta
genome.all.maker.snap_masked.transcripts.fasta
genome.all.maker.transcripts.fasta
genome.all.maker.trnascan.noncoding.fasta
maker_bopts.ctl
maker_evm.ctl
maker_exe.ctl
maker.gff
maker_opts.ctl
maker_stat.txt
protein2genome.gff
protein_gff_protein2genome.gff
repeat_gff_repeatmasker.gff
repeat_gff_repeatrunner.gff
snap_masked.gff
tblastx.gff
</code></pre>

<p>Here is a describtion of the most important files:</p>
<ul>
<li><strong>maker.gff</strong> </li>
</ul>
<p>It contains the annotation done by maker in (<a href="http://www.sequenceontology.org/gff3.shtml">GFF3 format</a>). If you use 'less' to read this annotation file, you will see a range of different features:</p>
<pre><code>##gff-version 3  
4       maker   gene    24134   25665   .       +       .       ID=maker-4-exonerate_protein2genome-gene-0.0;Name=maker-4-exonerate_protein2genome-gene-0.0
4       maker   mRNA    24134   25665   917     +       .       ID=maker-4-exonerate_protein2genome-gene-0.0-mRNA-1;Parent=maker-4-exonerate_protein2genome-gene-0.0;Name=maker-4-exonerate_protein2genome-gene-0.0-mRNA-1;_AED=0.09;_eAED=0.09;_QI=0|0.33|0.25|1|0|0|4|44|290
</code></pre>

<p>...</p>
<p>For example, the above lines read:</p>
<p>On the sequence with id ´4´, there is a gene feature located from position 24134 to 25665, on the plus strand and with the id 'maker-4-exonerate_protein2genome-gene-0.0'. 
On this same sequence, belonging to the gene, is located a transcript from position 24134 to 25665, on the plus strand and with the id 'maker-4-exonerate_protein2genome-gene-0.0-mRNA-1'. It's quality, or AED score, is 0.09 - which means that the evidence alignments are close to be in perfect agreement with the transcript model.
And so on.</p>
<ul>
<li>
<p><strong>maker_stat.txt</strong><br />
Statistics of the annotation (maker.gff annotation file).</p>
</li>
<li>
<p><strong>genome.all.maker.transcripts.fasta</strong><br />
This fasta file contains the nucleotide sequences of the transcripts (mRNA) of the MAKER gene models.</p>
</li>
<li>
<p><strong>genome.all.maker.proteins.fasta</strong> 
This fasta file contains the amino acid sequences of the proteins translated from the CDS of the MAKER gene models.</p>
</li>
<li>
<p><strong>cdna2genome.gff</strong><br />
Contains the fasta sequences provided by the <strong>est=</strong> parameter that have been succefuly mapped onto the assembly.</p>
</li>
<li>
<p><strong>est_gff_xxx_.gff</strong><br />
Contains the gff data provided by the <strong>est_gff=</strong> parameter that have been kept.</p>
</li>
<li>
<p><strong>protein2genome.gff</strong><br />
Contains the fasta sequences provided by the <strong>protein=</strong> parameter that have been succefuly mapped onto the assembly.</p>
</li>
<li>
<p><strong>protein_gff_xxx_.gff</strong><br />
Contains the gff data provided by the <strong>protein_gff=</strong> parameter that have been kept.</p>
</li>
<li>
<p><strong>blastx.gff</strong><br />
Contains the blast results of the protein fasta sequences provided by the <strong>protein=</strong> parameter.</p>
</li>
<li>
<p><strong>blastn.gff</strong><br />
Contains the blast results of the nucleotide sequences provided by the <strong>est=</strong> parameter.</p>
</li>
<li>
<p><strong>tblastx.gff</strong><br />
Contains the blast results of the nucleotide sequences provided by the <strong>alt_est=</strong> parameter.</p>
</li>
<li>
<p><strong>repeatmasker.gff</strong><br />
Contains the repeat masking results when <strong>model_org=</strong>  or/and  <strong>rmlib=</strong>  parameter is used.</p>
</li>
<li>
<p><strong>repeatrunner.gff</strong><br />
Contains the repeatrunner results when <strong>repeat_protein=</strong> parameter is used.</p>
</li>
<li>
<p><strong>genome.all.maker.trnascan.noncoding.fasta</strong><br />
Contains the result in fasta format of the tRNAscan analysis when the parameter <strong>trna=1</strong> parameter is used. The corresponding gff result is within the maker.gff file.</p>
</li>
<li>
<p><strong>augustus_masked.gff</strong><br />
Contains the raw augustus annotation (not filter by MAKER) un gff format.</p>
</li>
<li>
<p><strong>genome.all.maker.augustus_masked.transcripts.fasta</strong><br />
Contains the transcripts of the raw augustus annotation (not filter by MAKER) in fasta format.</p>
</li>
<li>
<p><strong>genome.all.maker.augustus_masked.proteins.fasta</strong><br />
Contains the proteins of the raw augustus annotation (not filter by MAKER) in fasta format (translated CDS).</p>
</li>
<li>
<p><strong>maker_xxx.ctl</strong><br />
All the file with <strong>.ctl</strong> extensions are copy of the control files used to produce the result contained in the current folder.</p>
</li>
</ul></div>
        
        
    </div>

    <footer class="col-md-12 text-center">
        
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
        
        
    </footer>

    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <!-- <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script> -->
    <script src="../../../js/base.js"></script>
    <script src="../../../search/main.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
