<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">  
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    
    <title>Practical 2 - Bioinformatics course</title>
    

    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/3.003/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="../../../css/base.min.css" rel="stylesheet">
    <link href="../../../css/cinder.min.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.min.css">
    <link href="../../../css/admonition.css" rel="stylesheet">
    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
            <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
        <![endif]-->

    <script src="//ajax.googleapis.com/ajax/libs/webfont/1.6.28/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    

     
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">First week <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../blast/blast_online/">Blast Online</a>
</li>

                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../blast/blast_cli/">Command-line Blast</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 2 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../software/">Installing Software</a>
</li>

                        
                            
<li >
    <a href="../../../git/">Git</a>
</li>

                        
                            
<li >
    <a href="../../../project_organisation/">Project Organisation</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/qc/">Quality Control</a>
</li>

                        
                            
<li >
    <a href="../../../assembly_challenge/">Assembly Challenge</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 4 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/rna/">RNA Sequencing</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/16S/">Metabarcoding</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/meta_assembly/">Metagenome assembly</a>
</li>

                        
                            
<li >
    <a href="../../../tutorials/docs/wms/">Comparative metagenomics</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 5 - Eukaryotic Annotation <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../schedule/">Schedule</a>
</li>

                        
                            
<li >
    <a href="../practical1/">Practical 1</a>
</li>

                        
                            
<li class="active">
    <a href="./">Practical 2</a>
</li>

                        
                            
<li >
    <a href="../practical3_manualCuration/">Practical 3</a>
</li>

                        
                            
<li >
    <a href="../practical4_funcAnnotInterp/">Practical 4</a>
</li>

                        
                            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Extras</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../UsingWebApollo.md">WebApollo</a>
</li>

        
            
<li >
    <a href="../practical2_sub_gatherEvidence/">Protein, EST and RNA-seq data</a>
</li>

        
            
<li >
    <a href="../practical2_sub_transcriptome/">Assembling transcripts</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerNoAbinit/">Evidence Based Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerAbinit/">Abinitio Annotation</a>
</li>

        
            
<li >
    <a href="../practical2_sub_makerCompareAnnot/">Comparing Annotations</a>
</li>

        
            
<li >
    <a href="../practical2_supl_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl3_maker/">Configuring Maker (I)</a>
</li>

        
            
<li >
    <a href="../practical2_supl2_maker/">Inspecting Maker output</a>
</li>

        
    </ul>
  </li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fas fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../practical1/">
                            <i class="fas fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../practical3_manualCuration/">
                            Next <i class="fas fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#gathering-evidence-data-for-annotation-optional">Gathering evidence data for annotation (Optional)</a></li>
            <li class="second-level"><a href="#1-protein-est-and-rna-seq-data-optional">1. Protein, EST and RNA-seq data (Optional)</a></li>
                
            <li class="second-level"><a href="#2-assembling-transcripts-based-on-rna-seq-data-optional">2. Assembling transcripts based on RNA-seq data (Optional)</a></li>
                
            <li class="second-level"><a href="#closing-remarks">Closing remarks</a></li>
                
        <li class="first-level "><a href="#running-the-maker-gene-build-pipeline">Running the Maker gene build pipeline</a></li>
            <li class="second-level"><a href="#overview">Overview</a></li>
                
            <li class="second-level"><a href="#understanding-makers-control-files">Understanding Makers control files</a></li>
                
            <li class="second-level"><a href="#running-maker-drosophila-genome">Running Maker - Drosophila genome</a></li>
                
                <li class="third-level"><a href="#1-creating-an-evidence-based-annotation">1. Creating an evidence based annotation</a></li>
                <li class="third-level"><a href="#2-creating-an-abinition-evidence-driven-annotation">2. Creating an abinition evidence-driven annotation</a></li>
                <li class="third-level"><a href="#3-inspecting-the-output">3. Inspecting the output</a></li>
            <li class="second-level"><a href="#closing-remarks_1">Closing remarks</a></li>
                
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="gathering-evidence-data-for-annotation-optional">Gathering evidence data for annotation (Optional)</h1>
<p>This exercise is meant to get you acquainted with the type of data you would normally encounter in an annotation project.</p>
<h2 id="1-protein-est-and-rna-seq-data-optional">1. Protein, EST and RNA-seq data (Optional)</h2>
<p><a href="../practical2_sub_gatherEvidence/">Here will get an idea of where to download evidence sequences.</a></p>
<h2 id="2-assembling-transcripts-based-on-rna-seq-data-optional">2. Assembling transcripts based on RNA-seq data (Optional)</h2>
<p><a href="../practical2_sub_transcriptome/">Here will get an idea how to deal with RNA-seq data</a></p>
<h2 id="closing-remarks">Closing remarks</h2>
<p>Now you know how to obtain evidence data, which will be useful to perform a nice annotation.</p>
<h1 id="running-the-maker-gene-build-pipeline">Running the Maker gene build pipeline</h1>
<h2 id="overview">Overview</h2>
<p><strong>MAKER</strong> is a computational pipeline to automatically generate annotations from a range of input data - including proteins, ESTs, RNA-seq transcripts and ab-initio gene predictions. During this exercise, you will learn how to use Maker with different forms of input data, and how to judge the quality of the resulting annotations.</p>
<p>The Maker pipeline can work with any combination of the following data sets:</p>
<ul>
<li>
<p>Proteins from the same species or related species  </p>
</li>
<li>
<p>Proteins from more distantly related organisms (e.g. Uniprot/Swissprot)  </p>
</li>
<li>
<p>EST sequences from the same species or very closely related species  </p>
</li>
<li>
<p>RNA-seq data from the same or very closely related species - in the form of splice sites or assembled transcripts  </p>
</li>
<li>
<p>Ab-initio predictions from one or more tools (directly supported are: Augustus, Snap, GeneMark, Fgenesh)  </p>
</li>
</ul>
<p>At minimum, most annotation projects will run with a protein data set, possibly complemented by some RNA-seq data. Popular examples of this are most of the traditional model systems, including human. However, a potential shortcoming of such approaches is that the comprehensiveness of the annotation depends directly on the input data. This can become a problem if our genome of interest is taxonomically distant to well-sequenced taxonomic groups so that only few protein matches can be found. Likewise, not all genes will be expressed at all times, making the generation of a comprehensive RNA-seq data set for annotation challenging.</p>
<p>We will therefore first run our annotation project in the traditional way, with proteins and ESTs, and then repeat the process with a well-trained ab-initio gene predictor. You can then compare the output to get an idea of how crucial the use of a gene predictor is. However, before we get our hands dirty, we need to understand Maker a little better...</p>
<p>Maker strings together a range of different tools into a complex pipeline (e.g. blast, exonerate, repeatmasker, augustus...), fortunately all its various dependencies have been already installed for you. 
Check that everything is running smoothly by creating the MAKER config files:</p>
<pre><code>cd ~/annotation_course/practical2
mkdir maker
cd maker
maker -CTL
</code></pre>

<h2 id="understanding-makers-control-files">Understanding Makers control files</h2>
<p>Makers behaviour and information on input data are specified in one of three control files. These are:</p>
<ul>
<li>maker_opts.ctl  </li>
<li>maker_bopts.ctl  </li>
<li>maker_exe.ctl</li>
</ul>
<p>What are these files for?</p>
<p>'maker_exe.ctl' holds information on the location of the various binaries required by Maker (including Blast, Repeatmasker etc). Normally, all information in this file will be extracted from $PATH, so if everything is set up correctly, you will never have to look into this file.</p>
<p>Next, 'maker_bopts.ctl' provides access to a number of settings that control the behaviour of evidence aligners (blast, exonerate). The default settings will usually be fine, but if you want to try to annotate species with greater taxonomic distance to well-sequenced species, it may become necessary to decrease stringency of the e.g. blast alignments.</p>
<p>Finally, 'maker_opts.ctl' holds information on the location of input files and some of the parameters controlling the decision making during the gene building.</p>
<h2 id="running-maker-drosophila-genome">Running Maker - Drosophila genome</h2>
<p>We will annotate the genome of the fruit fly <em>Drosophila melanogaster</em>. First we will perforn a pure evidence based annotation (without ab-initio predictions) and afterwards with ab-initio.</p>
<h3 id="1-creating-an-evidence-based-annotation">1. Creating an evidence based annotation</h3>
<p><a href="../practical2_sub_makerNoAbinit/">Running Maker with only evidence data</a></p>
<h3 id="2-creating-an-abinition-evidence-driven-annotation">2. Creating an abinition evidence-driven annotation</h3>
<p><a href="../practical2_sub_makerAbinit/">Running Maker with ab-initio predictions</a></p>
<h3 id="3-inspecting-the-output">3. Inspecting the output</h3>
<p>The running of an annotation pipeline like Maker is not actually very hard. But the complicated work is only beginning. How to we best inspect the gene builds? Count features? Visualize it? Most importantly, what steps do we need to take to create a 'finished' annotation that we can use for scientific analyses?</p>
<p><a href="../practical2_sub_makerCompareAnnot/">Comparing and evaluating annotations</a></p>
<h2 id="closing-remarks_1">Closing remarks</h2>
<p>This concludes the gene building part. We have learned how to use the Maker annotation pipeline and have created gene builds with and without ab-initio predictions. Moreover, we have employed some measures to describe and judge these annotations. An essential part that we decided to leave out is the training of ab-initio gene finders. The reason for this omission was that there isn't really any one best way to do this and your mileage may vary a lot based on your organism and input data. Perhaps the most direct approach available at the moment is a combination of evidence-based annotation with Maker and to use the resulting, crude gene models to train SNAP. Since Maker can improve ab-initio predictions 'on the fly', it can tolerate a bit of noise from a less-than-perfect ab-initio profile. If you are setting out on an annotation project, the NBIS annotation service would be happy to discuss the best approach for your data with you.</p>
<p>With that being said, generating a gene build is only one part of an annotation project. Next, we will inspect the annotation in genome browser and make an attempt at functional inference for the predicted gene models.</p></div>
        
        
    </div>

    <footer class="col-md-12 text-center">
        
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
        
        
    </footer>

    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <!-- <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script> -->
    <script src="../../../js/base.js"></script>
    <script src="../../../search/main.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
