<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">   
    <link rel="shortcut icon" href="../../../img/favicon.ico">

    <title></title>

    <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/2.018/css/hack.min.css">
    <link href='//fonts.googleapis.com/css?family=PT+Sans:400,400italic,700,700italic&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,700italic,400,300,600,700&subset=latin-ext,latin' rel='stylesheet' type='text/css'>
    <link href="../../../css/base.css" rel="stylesheet">
    <link href="../../../css/cinder.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">


    <link href="../../../css/admonition.css" rel="stylesheet">


    <link href="../../../css/navbar.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
    <script>
    WebFont.load({
        google: {
            families: ['Open Sans', 'PT Sans']
        }
    });
    </script>

    
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->

            <a class="navbar-brand" href="../../..">Bioinformatics course</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Home</a>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 1 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../proteins/">Proteins</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 2 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../unix/">Unix</a>
</li>

                        
                            
<li >
    <a href="../../../cloud/">Cloud Computing</a>
</li>

                        
                            
<li >
    <a href="../../../software/">Installing Software</a>
</li>

                        
                            
<li >
    <a href="../../../git/">Git</a>
</li>

                        
                            
<li >
    <a href="../../../blast/">Command-line Blast</a>
</li>

                        
                            
<li >
    <a href="../../../project_organisation/">Project Organisation</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Week 3 <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../../seq_tech/">Sequencing Technologies</a>
</li>

                        
                            
<li >
    <a href="../qc/">Quality Control</a>
</li>

                        
                            
<li class="active">
    <a href="./">Mapping and Variant Calling</a>
</li>

                        
                            
<li >
    <a href="../assembly/">De-novo Genome Assembly</a>
</li>

                        
                            
<li >
    <a href="../annotation/">Genome Annotation</a>
</li>

                        
                            
<li >
    <a href="../pan_genome/">Pan-genome Analysis</a>
</li>

                        
                            
<li >
    <a href="../nanopore/">Nanopore Sequencing</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>

            <ul class="nav navbar-nav navbar-right">
                    <li>
                        <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                            <i class="fa fa-search"></i> Search
                        </a>
                    </li>
                    <li >
                        <a rel="next" href="../qc/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../assembly/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="first-level active"><a href="#mapping-and-variant-calling">Mapping and Variant Calling</a></li>
            <li class="second-level"><a href="#downloading-a-reference">Downloading a Reference</a></li>
                
                <li class="third-level"><a href="#indexing-the-reference">Indexing the reference</a></li>
                <li class="third-level"><a href="#aligning-reads">Aligning reads</a></li>
                <li class="third-level"><a href="#visualising-with-tview">Visualising with tview</a></li>
                <li class="third-level"><a href="#variant-calling">Variant Calling</a></li>
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="mapping-and-variant-calling">Mapping and Variant Calling</h1>
<p>In this practical you will learn to map NGS reads to a reference sequence, check the output using a viewer software and investigate some aspects of the results.
You will be using the read data from the <a href="../qc/">Quality Control</a> practical.</p>
<p>EHEC O157 strains generally carry a large virulence plasmid, pO157.
Plasmids are circular genetic elements that many bacteria carry in addition to their chromosomes.
This particular plasmid encodes a number of proteins which are known or suspected to be involved in the ability to cause severe disease in infected humans.
Your task in this practical is to map your prepared read set to a reference sequence of the virulence plasmid, to determine if the pO157 plasmid is present in the St. Louis outbreak strain.</p>
<hr />
<p><img alt="plasmid" src="../img/plasmid.png" />
<em>Illustration of plasmid integration into a host bacteria</em></p>
<hr />
<h2 id="downloading-a-reference">Downloading a Reference</h2>
<p>You will need a reference sequence to map your reads to.</p>
<pre><code class="bash">cd ~/work
curl -O -J -L https://osf.io/rnzbe/download
</code></pre>

<p>This file contains the sequence of the pO157 plasmid from the Sakai outbreak strain of E. coli O157.
In contrast to the strain we are working on, this strain is available as a finished genome, i.e. the whole sequence of both the single chromosome and the large virulence plasmid are known.</p>
<h3 id="indexing-the-reference">Indexing the reference</h3>
<p>Before aligning the reads against a reference, it is necessary to build an index of that reference</p>
<pre><code class="bash">bowtie2-build pO157_Sakai.fasta.gz pO157_Sakai
</code></pre>

<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Indexing the reference is a necessary pre-processing step that makes searching for patterns much much faster. Many popular aligners such as Bowtie and BWA use an algorithm called the <a href="https://en.wikipedia.org/wiki/Burrows–Wheeler_transform">Burrows–Wheeler transform</a> to build the index.</p>
</div>
<h3 id="aligning-reads">Aligning reads</h3>
<p>Now we are ready to map our reads</p>
<pre><code class="bash">bowtie2 -x pO157_Sakai -1 SRR957824_trimmed_R1.fastq.gz \
    -2 SRR957824_trimmed_R2.fastq.gz -S SRR957824.sam
</code></pre>

<p>The output of the mapping will be in the SAM format.
You can find a brief explanation of the SAM format <a href="file_formats.md">here</a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In this tutorial as well as many other places, you'll often see the terms <em>mapping</em> and <em>alignment</em> being used interchangeably. If you want to read more about the difference between the two, I invite you to read this excellent <a href="https://www.biostars.org/p/180986/">Biostars discussion</a></p>
</div>
<h3 id="visualising-with-tview">Visualising with tview</h3>
<pre><code class="bash">head SRR957824.sam
</code></pre>

<p>But it is not very informative.
We'll use samtools to visualise our data</p>
<p>Before downloading the data in tablet, we have to convert our SAM file into BAM, a compressed version of SAM that can be indexed.</p>
<pre><code class="bash">samtools view -hSbo SRR957824.bam SRR957824.sam
</code></pre>

<p>Sort the bam file per position in the genome and index it</p>
<pre><code class="bash">samtools sort SRR957824.bam SRR2584857.sorted.bam
samtools index SRR2584857.sorted.bam
</code></pre>

<p>Finally we can visualise with <code>samtools tview</code></p>
<pre><code class="bash">samtools tview SRR2584857.sorted.bam pO157_Sakai.fasta.gz
</code></pre>

<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>navigate in tview:<br />
    - left and right arrows scroll  <br />
    - <code>q</code> to quit <br />
    - <code>CTRL-h</code> and <code>CTRL-l</code> scrolls more<br />
    - <code>g gi|10955266|ref|NC_002128.1|:8000</code> will take you to a specific location.   </p>
</div>
<h3 id="variant-calling">Variant Calling</h3>
<p>A frequent application for mapping reads is variant calling, i.e. finding positions where the reads are systematically different from the reference genome.
Single nucleotide polymorphism (SNP)-based typing is particularly popular and used for a broad range of applications.
For an EHEC O157 outbreak you could use it to identify the source, for instance.</p>
<p>We can call the variants using <code>samtools mpileup</code></p>
<pre><code class="bash">samtools mpileup -uD -f pO157_Sakai.fasta.gz SRR2584857.sorted.bam | \
    bcftools view - &gt; variants.vcf
</code></pre>

<p>You can read about the structure of vcf files <a href="file_formats.md">here</a>.
The documentation is quite painful to read and take a look at the file</p>
<p>Look at the non-commented lines</p>
<pre><code class="bash">grep -v ^## variants.vcf
</code></pre>

<p>The first five columns are <em>CHROM POS ID REF ALT</em>.</p>
<p>Use</p>
<pre><code class="bash">grep -v ^## variants.vcf | less -S
</code></pre>

<p>for a better view.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Use your left and right arrows to scroll horizontally, and <code>q</code> to quit.</p>
</div>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>How many SNPs did the variant caller find? Did you find any indels?</p>
</div>
<p>Examine one of the variants with tview</p>
<pre><code class="bash">samtools tview SRR2584857.sorted.bam pO157_Sakai.fasta.gz \
    -p 'gi|10955266|ref|NC_002128.1|:43071'
</code></pre>

<p>That seems very real!</p>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Where do reference genomes come from?</p>
</div></div>
        
    </div>

    <footer class="col-md-12 text-center">
        <hr>
        <p>
        <small>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></small>
    </footer>

    <script src="../../../js/jquery-1.10.2.min.js"></script>
    <script src="../../../js/bootstrap-3.0.3.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script>
    var base_url = '../../..';
    </script>
    <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
    <script src="../../../js/base.js"></script>
    <script src="../../../search/require.js"></script>
    <script src="../../../search/search.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
                <div class="modal-footer">
                </div>
            </div>
        </div>
    </div>

    </body>

</html>
